hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/join/JoinPredicatePreparedStatementSetter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.jdbc.join;
+
+import com.hazelcast.internal.util.ExceptionUtil;
+import com.hazelcast.jet.sql.impl.JetJoinInfo;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class JoinPredicatePreparedStatementSetter implements Consumer<PreparedStatement> {
+
+    private final JetJoinInfo joinInfo;
+
+    private final List<JetSqlRow> leftRowsList;
+
+    public JoinPredicatePreparedStatementSetter(JetJoinInfo joinInfo, List<JetSqlRow> leftRowsList) {
+        this.joinInfo = joinInfo;
+        this.leftRowsList = leftRowsList;
+    }
+
+    @Override
+    public void accept(PreparedStatement preparedStatement) {
+        try {
+            setObjectsToPreparedStatement(preparedStatement);
```

Comment:
I tried to refactor it and get rid of the redundant (IMO) method. 
I prepared the code below, but won't force you to refactor existing code, just proceed with what you like more.
IMO, the version below is more readable.

```
@Override
public void accept(PreparedStatement preparedStatement) {
    int[] leftEquiJoinIndices = joinInfo.leftEquiJoinIndices();

    // PreparedStatement parameter index starts from 1
    int parameterIndex = 1;

    // leftRow contains all left table columns used in the select statement
    // leftEquiJoinIndices contains index of columns used in the JOIN clause
    try {
        for (JetSqlRow leftRow : leftRowsList) {
            for (int leftEquiJoinIndexValue : leftEquiJoinIndices) {
                Object value = leftRow.get(leftEquiJoinIndexValue);
                preparedStatement.setObject(parameterIndex++, value);
            }
        }
    } catch (SQLException e) {
        ExceptionUtil.sneakyThrow(e);
    }
}
```

Reply:
If you find it more readable why not 
rev : ee4dd644caf2ae558cd05c5ed01432577fb34ac6